tests/sles4sap/publiccloud/qesap_terraform.pm

```diff
@@ -141,7 +141,7 @@ sub run {
         $instance->wait_for_ssh();
         # Does not fail for some reason.
         my $real_hostname = $instance->run_ssh_command(cmd => "hostname", username => "cloudadmin");
-        die if ($expected_hostname ne $real_hostname);
+        die if ($expected_hostname ne $real_hostname && lc(get_required_var('PUBLIC_CLOUD_PROVIDER')) ne 'ec2');
```

Comment:
Why is this condition excluded on EC2 ?

Reply:
> Why is this hostname condition excluded on EC2 ?

The instance name is not "vmhana01" likes on Azure but "https://openqaworker15.qa.suse.cz/tests/94765#step/qesap_terraform/267" (for example) on EC2.

Reply:
Ok, then IMO this information should be part of, at least, commit message, because your commit message does not say anything about the change itself only that is part of ticket.

Reply:
> Ok, then IMO this information should be part of, at least, commit message, because your commit message does not say anything about the change itself only that is part of ticket.

The commit message revised. Thanks for the good question.

Reply:
Hmmm... this is a good point here. It means that deployment solution (or something in perl code?) returns "vmhana0*" as  but on the EC2 host the hostname is different. 
I am thinking that maybe this could be labeled as a workaround with "record_soft_failure" and take a look into this at least to know why is there such discrepancy.
I will create a jira ticket for that.

Reply:
> > Why is this hostname condition excluded on EC2 ?
> 
> The instance name is not "vmhana01" likes on Azure but "https://openqaworker15.qa.suse.cz/tests/94765#step/qesap_terraform/267" (for example) on EC2.

I like to see how it is the Ansible inventory generated by terraform step, but I'm not able to find it https://openqaworker15.qa.suse.cz/tests/94765#downloads

Reply:
Thanks for the `hostname` comments.

Reply:
> I am actually wondering if we need that `die` there in the first place (without an error message, no less). We have known for a while that on EC2 we do not get the actual hostnames in the terraform output as in the other public cloud providers, so the check makes no sense to me.
> 
> If keeping it, I would recommend changing that to make it more explicit what we are doing. My suggestion would be to remove L144 and replace it with:
> 
> ```
> # We expect hostnames reported by terraform to match the actual hostnames in Azure and GCE
> die "Expected hostname $expected_hostname is different than actual hostname [$real_hostname]"
>     if ((is_azure || is_gce) && ($expected_hostname ne $real_hostname));
> ```

Done. Thanks for the example. The according VRs are passed:
    EC2: http://openqaworker15.qa.suse.cz/tests/96415#step/qesap_terraform/267
    Azure: https://openqaworker15.qa.suse.cz/tests/96414#step/qesap_terraform/327